odoo/models.py

```diff
@@ -2246,11 +2246,17 @@ def _read_group_prepare(self, orderby, aggregated_fields, annotated_groupbys, qu
             for key in ('field', 'groupby')
         }
         for order_part in orderby.split(','):
-            order_split = order_part.split()
+            order_split = order_part.split()  # potentially ["field:agg", "desc"]
             order_field = order_split[0]
+            if '.' in order_field:
+                order_field, is_many2one_id = order_field.split(".")
+            else:
+                is_many2one_id = None
             if order_field == 'id' or order_field in groupby_fields:
-                if self._fields[order_field.split(':')[0]].type == 'many2one':
-                    order_clause = self._generate_order_by(order_part, query).replace('ORDER BY ', '')
+                order_field_without_aggregate = order_field.split(':')[0]
+                if self._fields[order_field_without_aggregate].type == 'many2one' and not is_many2one_id:
```

Comment:
The long name doesn't help for reading the condition.  What about this?
```suggestion
                field = self._fields[order_field.split(':')[0]]
                if field.type == 'many2one' and not is_many2one_id:
```
As a side note, nowadays I prefer to name `field` the instances of `Field`, and their name `name`, `fname` or `field_name`.  But this code does not follow this convention, so it's up to you...